### ISSUE TITLE
Offscreen windows have wrong viewport size

### ISSUE THREAD

**Issue by Mr0grog** [electron/electron#8224]
* Electron version: 1.4.12
* Operating system: Mac OS X 10.11.6

When rendering using offscreen mode, the viewport/render buffer is the size of the window instead of the window’s content area. I believe this is because `OffScreenRenderWidgetHostView` takes its size from its native window’s`GetSize` method and not `GetContentSize`:
- https://github.com/electron/electron/blob/master/atom/browser/osr/osr_render_widget_host_view.cc#L350 and
- https://github.com/electron/electron/blob/master/atom/browser/osr/osr_render_widget_host_view.cc#L929

Here’s a simple test case to reproduce the issue:

```js
const electron = require('electron');

electron.app.on('ready', function() {
  const measurements = {normal: null, offscreen: null, offscreenBuffer: null};
  
  function saveMeasurement (measurement, value) {
    if (measurement && value) {
      measurements[measurement] = value;
    }
    
    if (measurements.normal && measurements.offscreen && measurements.offscreenBuffer) {
      console.log(`Normal viewport size:    ${measurements.normal}`);
      console.log(`Offscreen viewport size: ${measurements.offscreen}`);
      console.log(`Offscreen buffer size:   ${measurements.offscreenBuffer}`);
      process.exit();
    }
  }
  
  function measureWindowSize (type) {
    const window = new electron.BrowserWindow({
      width: 500,
      height: 500,
      webPreferences: {
        offscreen: type === 'offscreen'
      }
    });
    
    window.webContents.on('did-finish-load', () => {
      window.webContents.executeJavaScript('[window.innerWidth, window.innerHeight]', false)
        .then(size => saveMeasurement(type, size));
    });
    
    window.webContents.once('paint', (event, dirty, image) => {
      const {width, height} = image.getSize()
      saveMeasurement('offscreenBuffer', [width, height]);
    });
    
    window.loadURL('about:blank');
  }
  
  measureWindowSize('normal');
  measureWindowSize('offscreen');
});
```

This logs:

```
Normal viewport size:    500,478
Offscreen viewport size: 500,500
Offscreen buffer size:   500,500
```

Each of those sizes *should* be same (`500,478` in this case, but that will vary by platform).

**Comment by gerhardberger** [comment]
This is expected, because an offscreen window behaves like a [frameless](https://github.com/electron/electron/blob/master/docs/api/frameless-window.md) window, because it cannot have a frame like traditional browser windows do.

**Comment by Mr0grog** [comment]
> This is expected, because an offscreen window behaves like a frameless window

If that's the case, shouldn’t I get the same results for off- vs. on-screen if I instantiate the window with `useContentSize`? e.g:

```js
const window = new electron.BrowserWindow({
  width: 500,
  height: 500,
  useContentSize: true,
  webPreferences: {
    offscreen: type === 'offscreen'
  }
});
```

In that case, the measurements still differ by the size of the window chrome.

**Comment by MarshallOfSound** [comment]
`useContentSize` still has the window Chrome, @gerhardberger said it behaves like a frameless window so try setting `frame: false`

**Comment by Mr0grog** [comment]
@MarshallOfSound A frameless window doesn’t differentiate between the content size and the size. So creating a window like this would get a viewport size of `500,500`, regardless of whether `useContentSize` is `true` or `false`:

```js
const window = new electron.BrowserWindow({
  width: 500,
  height: 500,
  frame: false,
  useContentSize: // either `true` or `false`
});
```

So if an offscreen window is the the same as a frameless window, this should give the same results:

```js
const window = new electron.BrowserWindow({
  width: 500,
  height: 500,
  frame: false,
  useContentSize: // either `true` or `false`
});
```

…but it doesn’t. If `useContentSize` is true, you get `500,522` and `500,500` otherwise. That seems like a bug to me. Otherwise, if that oddity in the behavior is expected, it should at least be documented in the [offscreen guide](https://github.com/electron/electron/blob/master/docs/tutorial/offscreen-rendering.md).

**Comment by Mr0grog** [comment]
Ugh, copy/paste error in the second example above, sorry. I’ve updated it (in case you are following by e-mail).

**Comment by Mr0grog** [comment]
(In a related note, the preview window you get on screen for the offscreen window includes chrome, but is still the specified size, so it clips the preview.)

**Comment by gerhardberger** [comment]
@Mr0grog `useContentSize` shouldn't affect the size of the offscreen window, so that's a bug indeed. Thanks for noticing it, I will try to look into it tomorrow!

**Comment by gerhardberger** [comment]
@Mr0grog I opened a PR, that creates the browser window frameless in offscreen mode, so the size of the window will be consistent, always the value you passed in as parameters.

**Comment by Mr0grog** [comment]
Awesome, thanks!

**Comment by gerhardberger** [comment]
Fixed in #8250
